MzML/MzmlMethods.cs

```diff
@@ -825,21 +968,144 @@ public static void CreateAndWriteMyMzmlWithCalibratedSpectra(IMsDataFile<IMsData
                 }
             }
 
-            if (writeIndexed)
-                throw
+            if (!writeIndexed)
+            {
+                using (TextWriter writer = new StreamWriter(outputFile))
+                {
 
-new MzLibException("Writing indexed mzMLs not yet supported");
-            else
+                    mzmlSerializer.Serialize(writer, mzML);
+                }
+            }
+
+            else if (writeIndexed)
             {
-                using (TextWriter writer = new StreamWriter(outputFile)
+                Generated.indexedmzML indexedMzml = new Generated.indexedmzML();
+
+                var inMemoryTextWriter = new System.IO.MemoryStream();
+
 
-)
+                //compute total offset
+                indexedMzml.mzML = mzML;
+
+
+                indexedSerializer.Serialize(inMemoryTextWriter, indexedMzml);
+                string allmzMLData = Encoding.UTF8.GetString(inMemoryTextWriter.ToArray()).Replace("\r\n", "\n");
```

Comment:
I don't think it can be done directly from the serializer. There is another way where you create an XmlTextwriter object then assign settings to it and then serialize into that object,  but I believe the current way is faster.